magento/magento2
# file upload issue in WYSIWYG editor on product page

### Title 
The folder-tree are not visible and getting error “We can't upload the file to the current folder right now. Please try another folder.” when we try to upload image 

### Description (*)

There is an issue encountered when attempting to upload an image to a folder within the WYSIWYG editor on a product page, specifically within the short description in content section. The folders are not visible, and as a result, we cannot upload the image and we receive an error message stating, 'We can't upload the file to the current folder right now. Please try another folder.' This problem only occurs when you expand the 'Image and Video' tab first. And getting the error in console  as shown in below image
 
![image](https://github.com/magento/magento2/assets/124759248/400f27cc-b770-416a-a914-af3cc6f63830)



### Manual testing scenarios (*)
1: Access the Magento admin panel and navigate to 'Catalog' -> 'Products.' and  Select any product.

2: in the product page, first, expand the 'Image and Videos' tab, as shown in the image below. 

![image](https://github.com/magento/magento2/assets/124759248/8dfb9118-d82f-4b4f-b890-6c75f6093afe)

3: Next, expand the 'Content' tab and in 'Short Description' section click on the 'Insert Image' button and select the 'Source' button in the WYSIWYG editor, as indicated in the image below

![image](https://github.com/magento/magento2/assets/124759248/9f3a2222-a283-43ff-9a67-af0158582ba2)

4: after clicking the 'Source' button, you may notice that the folders are not displayed, and as a result, you are unable to upload images within the 'Short Description.' as you can see in below image

![image](https://github.com/magento/magento2/assets/124759248/cbd4cf72-ce51-41c9-ace7-061d53113bc9)

### **Expected result **
After expanding image and videos tab first we should be able to see folder option and can insert image in short description

### **Actual result **
After expanding image and videos tab first we can not see folder option and can not insert image in short-description 

### **Aditional information**
In this issue, when we expand the 'Image and Videos' tab and then attempt to upload an image within the 'Short Description' content section, it triggers the 'tmp.init(el,options);'  in the lib/web/jquery/jstree/jquery.jstree.js file at line 123, as illustrated in the image below.

![image](https://github.com/magento/magento2/assets/124759248/4f75b7e5-b7d7-4360-be9a-a49b1950f22e)

After the 'tmp.init(el, options)' function is triggered in the 'jquery.jstree.js' file, it call the init function in the 'tree-suggest.js' file. Please note that it call init function in  the 'tree-suggest.js' file  only when we initially expand the 'Image and Videos' tab first.

In the lib/web/mage/backend/tree-suggest.js file, it calls the ‘init’ function at line 58 however, an issue arises because the ‘init’ function does not receive the element. Consequently, ‘this.get_container’ becomes undefined 

This issue likely plays a significant role in the failure to display folders in the WYSIWYG editor when attempting to upload an image.the element contains the data of folder-tree so if it not assign correct we can not see folder tree when we try to insert image using  WYSIWYG editor in short-description

![image](https://github.com/magento/magento2/assets/124759248/48fdf578-f740-4a60-b976-939449ff77f4)
In the image provided above, you can see on line 77 and also on line 102, there is a call to the '_get_node' method. However, there isn't a method with this name in the file or jquery.jstree.js file. Instead, the correct method name is 'get_node.' Therefore, we need to make this correction to ensure the proper function is called.

### **Solution**

To resolve the issue, we need to make changes in the 'lib/web/mage/backend/tree-suggest.js' file. Specifically, we should pass the 'element' in the 'init' function at line 58. Afterward, we should call the 'init' function before 'get_container' and pass the 'data' (consisting of 'element' and 'option') to ensure that the 'init' function receives the required element. This modification ensures that the 'container' variable is not undefined. 

Additionally, we should replace instances of '_get_node' with 'get_node' on lines 77 and 102 to ensure that the correct method is called.

Once these modifications are implemented, the issue should be successfully resolved.you can see changes in below image

![image](https://github.com/magento/magento2/assets/124759248/925b3ae8-89cc-498f-b0af-07a2a51c681a)

### Questions or comments


### Contribution checklist (*)


### Resolved issues:
1. [x] resolves magento/magento2#38026: file upload issue in WYSIWYG editor on product page
